boa_engine/src/vm/mod.rs

```diff
@@ -209,260 +297,80 @@ impl Context<'_> {
     }
 
     pub(crate) fn run(&mut self) -> CompletionRecord {
-        #[cfg(feature = "trace")]
-        const COLUMN_WIDTH: usize = 26;
-        #[cfg(feature = "trace")]
-        const TIME_COLUMN_WIDTH: usize = COLUMN_WIDTH / 2;
-        #[cfg(feature = "trace")]
-        const OPCODE_COLUMN_WIDTH: usize = COLUMN_WIDTH;
-        #[cfg(feature = "trace")]
-        const OPERAND_COLUMN_WIDTH: usize = COLUMN_WIDTH;
-        #[cfg(feature = "trace")]
-        const NUMBER_OF_COLUMNS: usize = 4;
-
         let _timer = Profiler::global().start_event("run", "vm");
 
         #[cfg(feature = "trace")]
         if self.vm.trace {
-            let msg = if self.vm.frames.last().is_some() {
-                " Call Frame "
-            } else {
-                " VM Start "
-            };
-
-            println!(
-                "{}\n",
-                self.vm
-                    .frame()
-                    .code_block
-                    .to_interned_string(self.interner())
-            );
-            println!(
-                "{msg:-^width$}",
-                width = COLUMN_WIDTH * NUMBER_OF_COLUMNS - 10
-            );
-            println!(
-                "{:<TIME_COLUMN_WIDTH$} {:<OPCODE_COLUMN_WIDTH$} {:<OPERAND_COLUMN_WIDTH$} Top Of Stack\n",
-                "Time",
-                "Opcode",
-                "Operands",
-            );
+            self.trace_call_frame();
         }
 
-        let current_stack_length = self.vm.stack.len();
-        self.vm
-            .frame_mut()
-            .set_frame_pointer(current_stack_length as u32);
-
-        // If the current executing function is an async function we have to resolve/reject it's promise at the end.
-        // The relevant spec section is 3. in [AsyncBlockStart](https://tc39.es/ecma262/#sec-asyncblockstart).
-        let promise_capability = self.vm.frame().promise_capability.clone();
-
-        let execution_completion = loop {
+        let mut result = Ok(CompletionType::Normal);
```

Comment:
Is there a specific benefit here to using the mutable result value vs. a scoped value?

Reply:
> Is there a specific benefit here to using the mutable result value vs. a scoped value?

This was done because if we throw a `JsResult` error then we have to call `ReThrow::evaluate()` then we have to do the same thing we do when we execute an instruction, so instead of duplicate code, we just set result and continue to handle it.

Reply:
I inlined `ReThrow` since it's small and a lot of the logic is in `handle_exception_at`, and allows us to remove a check for generator return too, made `result` immutable :)